#! /bin/sh /usr/share/dpatch/dpatch-run
## 30_fix_build_for_debian.dpatch by Robin Cornelius <robin.cornelius@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Dont apply the -s strip flag to the shared lib debian will do it for us 
## DP: where necessary
## DP: Remove unnecessary linkage on lstdc++ and lm and add missing linkage for -lm

@DPATCH@

Index: ./Makefile
===================================================================
--- ./Makefile	(revision 96)
+++ ./Makefile	(working copy)
@@ -18,8 +18,7 @@
 # Converts cr/lf to just lf
 DOS2UNIX = dos2unix
 
-COMPILERFLAGS = -Wall -O3 -ffast-math -std=c99 -fPIC
-LIBRARIES = -lstdc++
+COMPILERFLAGS = -Wall -O3 -ffast-math -std=c99 -fPIC -g $(EXTRA_CFLAGS)
 
 MODULES = $(SRCS:.c=.o)
 CFLAGS = $(COMPILERFLAGS) $(INCLUDE)
@@ -54,7 +53,7 @@
 	$(AR) r $@ $(MODULES)
 
 $(SHAREDLIB): $(MODULES)
-	$(CC) -s -shared -Wl,-soname,$(LIBNAME) -o $@ $(MODULES) $(LIBRARIES)
+	$(CC) -shared -Wl,-soname,$(LIBNAME) -o $@ $(MODULES) -lm
 
 install: OpenJPEG
 	install -d '$(DESTDIR)$(INSTALL_LIBDIR)' '$(DESTDIR)$(INSTALL_INCLUDE)'

Index: ./indexer_JPIP/Makefile
===================================================================
--- ./indexer_JPIP/Makefile	(revision 95)
+++ ./indexer_JPIP/Makefile	(working copy)
@@ -1,7 +1,6 @@
 CC = gcc
 
-LDFLAGS = -lm
-CFLAGS = -Wall
+CFLAGS = -Wall $(EXTRA_CFLAGS)
 
 all: index_create
 
Index: ./codec/Makefile
===================================================================
--- ./codec/Makefile	(revision 96)
+++ ./codec/Makefile	(working copy)
@@ -1,13 +1,13 @@
 # Makefile for the main OpenJPEG codecs: j2k_to_image and image_to_j2k
 
-CFLAGS = -O3 -lstdc++ # -g -p -pg
+CFLAGS = $(EXTRA_CFLAGS) # -g -p -pg
 
 all: j2k_to_image image_to_j2k
 
-j2k_to_image: j2k_to_image.c ../libopenjpeg.a
+j2k_to_image: j2k_to_image.c
 	gcc $(CFLAGS) compat/getopt.c index.c convert.c j2k_to_image.c -o j2k_to_image -L.. -lopenjpeg -I ../libopenjpeg/ -lm -ltiff
 
-image_to_j2k: image_to_j2k.c ../libopenjpeg.a
+image_to_j2k: image_to_j2k.c
 	gcc $(CFLAGS) compat/getopt.c index.c convert.c image_to_j2k.c -o image_to_j2k -L.. -lopenjpeg -I ../libopenjpeg/ -lm -ltiff
 
 clean:
Index: ./mj2/Makefile
===================================================================
--- ./mj2/MakeFile	2008-03-10 08:50:35.000000000 +0000
+++ ./mj2/Makefile	2009-04-21 11:47:59.000000000 +0100
@@ -1,6 +1,6 @@
 # Makefile for the MJ2 codecs of the OpenJPEG library: frames_to_mj2, mj2_to_frames, extract_j2k_from_mj2 and wrap_j2k_in_mj2
 
-CFLAGS = -O3 -lstdc++ # -g -p -pg
+CFLAGS = ${EXTRA_CFLAGS} # -O3 -lstdc++ # -g -p -pg
 
 all: frames_to_mj2 mj2_to_frames extract_j2k_from_mj2 wrap_j2k_in_mj2
 
